### PR TITLE
Ava 233/scripts

### PR DESCRIPTION
**Pull request by kroos47**

### CODE REVIEW

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+func submit_data( size int) {
+
+	// This sample shows how to create a transaction to make a Avail data submission
+
+	// Instantiate the API (locally)
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
+	// api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:21]: nit: delete line

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	rv, err := api.RPC.State.GetRuntimeVersionLatest()
+	if err != nil {
+		panic(err)
+	}
+
+	key, err := types.CreateStorageKey(meta, "System", "Account", signature.TestKeyringPairAlice.PublicKey)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:53]: Can we take the keypair from config?

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+func submit_data( size int) {
+
+	// This sample shows how to create a transaction to make a Avail data submission
+
+	// Instantiate the API (locally)
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:20]: Does this use the URL from config?

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		panic(err)
+	}
+
+	fmt.Printf("Data submitted by Alice: %v against appID %v\n", data, appID)
+}
+
+
+func main() {
+	data_size := flag.Int("size",100, "size of data submission")
+	data:= *data_size
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:94]: nit: spacing

File: tests/gsrpc/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+go to the gsrpc folder in tests 
+
+```
+cd avail/tests/gsrpc
+```
+
+> Using simple data submission 
+
+``` 
+go run extrinsics/dataSubmission/dataSubmission.go -- -size 1000
```
Comment (prabal-banerjee) [inline tests/gsrpc/README.md:28]: Could not make it work with the config

File: tests/gsrpc/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+The same procedure of data submission is done here. But checks/logs if the data is included in the block. The default APP_ID is 0
+
+>LoadTesting(only for testing)
+
+```
+go run extrinsics/loadTestingTool/loadTestingTool.go -config config.json
+```
+Load testing the network with a random blob of 8000bytes data for the 10secs 
+
+
```
Comment (prabal-banerjee) [inline tests/gsrpc/README.md:53]: nit: needs formatting and spelling check.

File: tests/gsrpc/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+```json
+{
+    "seed":"bottom drive obey lake curtain smoke basket hold race lonely fit walk//Alice",
+    "api_url":"ws://127.0.0.1:9944",
+    "size":1000
+}
+```
+
+2. There are multiplt scripts you have in gsrpc
```
Comment (prabal-banerjee) [inline tests/gsrpc/README.md:17]: typo: `multiple`

File: tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	}
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		panic(err)
+	}
+
+	// Set data and appID according to need
+	data , _:=  extrinsics.RandToken(size)
+	appID := 0
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmission/dataSubmission.go:33]: possible fetch `appID` from config?

File: tests/gsrpc/extrinsics/blockListener/blockListener.go [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+	}
+
+	var accountInfo types.AccountInfo
+	ok, err := api.RPC.State.GetStorageLatest(key, &accountInfo)
+	if err != nil || !ok {
+		panic(err)
+	}
+	nonce := uint32(accountInfo.Nonce)
+	log.Println("Nonce: ", nonce)
+	data, _:= extrinsics.RandToken(config.Size)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/blockListener/blockListener.go:73]: nit: spacing

File: tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+func main() {
+	// This sample shows how to create a transaction to make a Avail data submission
+	data_size := flag.Int("size",100, "size of data submission")
+	data:= *data_size
+	flag.Parse()
+	// Instantiate the API (locally)
+	api, err := gsrpc.NewSubstrateAPI(config.Default().RPCURL)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/dataSubmitAndWatch/dataSubmitAndWatch.go:21]: Could not make it read from config.

File: tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+	log.Println("Subscribed to new headers...")
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		panic(err)
+	}
+
+	// if testing locally with Alice account, use signature.TestKeyringPairAlice.PublicKey as last param
+	// mneumonic for local Alice account: `bottom drive obey lake curtain smoke basket hold race lonely fit walk//Alice`
+	key, err := types.CreateStorageKey(meta, "System", "Account", signature.TestKeyringPairAlice.PublicKey)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go:54]: Could not make it work with custom mnemonic of config file.

File: tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	var accountInfo types.AccountInfo
+	ok, err := api.RPC.State.GetStorageLatest(key, &accountInfo)
```
Comment (prabal-banerjee) [inline tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go:60]: Got error here and it panics.

File: tests/gsrpc/internal/config/config.go [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type Config struct {
+	Seed   string `json:"seed"`
+	ApiURL string `json:"api_url"`
+	Size int `json:"size"`
+}
```
Comment (prabal-banerjee) [inline tests/gsrpc/internal/config/config.go:13]: Can we add `appID`?

**Review by prabal-banerjee — COMMENTED** [review]
Partial feedback on the `gsrpc` parts.

File: tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go (hunk 12 above)
Comment (kroos47) [reply tests/gsrpc/extrinsics/loadTestingTool/loadTestingTool.go:60]: modified some changes to take custom mnemonic. working now for me. Could you reverifiy after commit?

**Comment by luka-ethernal** [comment]
Since we now have avail-subxt in the repo, the existing rust scripts should either be updated or, since we have examples there, removed. If we don't remove them, let's make them distinct in terms of functionality.

**Review by luka-ethernal — APPROVED** [review]

**Pull request merged**